lib/sedna/core/federated_learning/federated_learning.py

```diff
@@ -43,76 +47,91 @@ def __init__(self, estimator, aggregation="FedAvg"):
         self.aggregation = ClassFactory.get_cls(ClassType.FL_AGG, aggregation)
+
+        connect_timeout = int(Context.get_parameters("CONNECT_TIMEOUT", "300"))
         self.node = None
+        self.register(timeout=connect_timeout)
 
-    def register(self):
+    def register(self, timeout=300):
         self.log.info(
             f"Node {self.worker_name} connect to : {self.config.agg_uri}")
         self.node = AggregationClient(
-            url=self.config.agg_uri, client_id=self.worker_name)
-        loop = asyncio.get_event_loop()
-        res = loop.run_until_complete(
-            asyncio.wait_for(self.node.connect(), timeout=300))
+            url=self.config.agg_uri,
+            client_id=self.worker_name,
+            ping_timeout=timeout
+        )
 
         FileOps.clean_folder([self.config.model_url], clean=False)
         self.aggregation = self.aggregation()
-        self.log.info(f"Federated learning Jobs model prepared -- {res}")
+        self.log.info(f"{self.worker_name} model prepared")
         if callable(self.estimator):
             self.estimator = self.estimator()
 
     def train(self, train_data,
               valid_data=None,
               post_process=None,
               **kwargs):
+        """
+        Training task for FederatedLearning
+        :param train_data: datasource use for train
+        :param valid_data: datasource use for evaluation
+        :param post_process: post process
+        :param kwargs: params for training of customize estimator
+        """
+
         callback_func = None
-        if post_process is not None:
+        if post_process:
             callback_func = ClassFactory.get_cls(
                 ClassType.CALLBACK, post_process)
 
         round_number = 0
         num_samples = len(train_data)
-        self.aggregation.total_size += num_samples
-
+        _flag = True
+        start = time.strftime("%Y-%m-%d %H:%M:%S", time.localtime())
+        res = None
         while 1:
-            round_number += 1
-            start = time.strftime("%Y-%m-%d %H:%M:%S", time.localtime())
-            self.log.info(
-                f"Federated learning start at {start},"
-                f" round_number={round_number}")
-            res = self.estimator.train(
-                train_data=train_data, valid_data=valid_data, **kwargs)
-
-            self.aggregation.weights = self.estimator.get_weights()
-            send_data = {"num_samples": num_samples,
-                         "weights": self.aggregation.weights}
-            received = self.node.send(
-                send_data, msg_type="update_weight", job_name=self.job_name)
-            exit_flag = False
-            if (received and received["type"] == "update_weight"
-                    and received["job_name"] == self.job_name):
-                recv = received["data"]
+            if _flag:
+                round_number += 1
+                start = time.strftime("%Y-%m-%d %H:%M:%S", time.localtime())
```

Comment:
don't need time here? since in general log has time already

Reply:
`start`  needs to be transmitted to LC as an element of `task_info`